Cloud vmware add guest customization options

ibuil opened the pull request:
### What does this PR do?

Based on VMware guest customization options in windows machines, we have added some features:
You can set a specific timezone, if it's not specified it sets GMT as default timezone for windows deployments.
You can add the machine to a specified Active Directory Domain
### Previous Behavior
We have found issues deploying Windows machinies as it sets a timezone different than the one configured at the template. Due to this problem we have to add an additional step during the deployment phase, although there is an option in the vmware api to do that.

### New Behavior
Via a variable in the profile or map file we can:
-  set specific timezones based on numeric values (https://msdn.microsoft.com/en-us/library/ms912391%28v=winembedded.11%29.aspx?f=255&MSPPError=-2147217396)
- integrate the deployed VM to Active Directory Windows Domain


### Tests written?

Yes/No
No


Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

kojiromike committed:
Avoid Early Convert ret['comment'] to String

Fixes this exception:
An exception occurred in this state: Traceback (most recent call last):
  File "/var/tmp/.syops_b1274f_salt/py2/salt/state.py", line 1733, in call
    **cdata['kwargs'])
  File "/var/tmp/.syops_b1274f_salt/py2/salt/loader.py", line 1652, in wrapper
    return f(*args, **kwargs)
  File "/var/tmp/.syops_b1274f_salt/py2/salt/states/gpg.py", line 119, in present
    ret['comment'].append('Adding {0} to GPG keychain'.format(name))
AttributeError: 'str' object has no attribute 'append'

dglloyd committed:
Pass sig to service.status in after_toggle

As on line 350, pass `sig` to `service.status` to ensure that the correct behavior is used. If we don't pass `sig`, upstart/sysv jobs that do not implement `service <name> status` will fail.

nicolas-geniteau committed:
Fix user creation with empty password

Emptying the password was only done if the user already existed, not
when creating.

Signed-off-by: Nicolas Geniteau <nicolas@cri.epita.net>

garethgreenaway committed:
Updating the git module to allow an identity file to be used when passing the user parameter

garethgreenaway committed:
Updating PR with requested changes.

skizunov committed:
Fix: Reactor emits critical error

Under normal usage, the reactor will emit this critical error:

```
[CRITICAL] kwargs must be passed inside the low data within the 'kwarg'
key. See usage of salt.utils.args.parse_input() and
salt.minion.load_args_and_kwargs() elsewhere in the codebase.
 ```

It seems like only `salt.utils.reactor.Reactor` uses `salt.state.Compiler`.
Due to this, it appears safe to customize `Compiler.compile_high_data`
for usage only by the reactor. Since reactor arguments are always named,
we ensure that each 'chunk' has an `arg` field that is an empty list
and a `kwarg` field that contains all the named arguments for use
with the given function call. This conforms to the format expected by
`salt.client.mixins.SyncClientMixin._low`.

Signed-off-by: Sergey Kizunov <sergey.kizunov@ni.com>

gtmanfred committed:
import salt.minion for EventReturn for Windows

Because of the way our multiprocessing works on Windows, this module is not
available.  In Linux, the namespace is copied over, and the module is
available, so we only need the import on Windows.

garethgreenaway committed:
Updating to swap on the new salt.utils.files.set_umask context_manager

garethgreenaway committed:
Fixing umask to we can set files as executable.

terminalmage committed:
Restrict set_umask to mkstemp call only

garethgreenaway committed:
Merge pull request #1 from terminalmage/pr-41945

Restrict set_umask to mkstemp call only

garethgreenaway committed:
Moving the call to cp.get_file inside the with block to ensure the umask is preserved when we grab the file.

terminalmage committed:
Prevent command from showing in exception when output_loglevel=quiet

When a command is being executed using salt.utils.timed_subprocess.TimedProc,
and that command does not exist in the PATH, it will result in an
OSError/IOError being raised. We catch this and then raise a
CommandExecutionError with some information, including the command being
run. However, when output_loglevel=quiet, we don't want any details of
the command to be visible.

This commit redacts the command details in these corner cases.

terminalmage committed:
Add debug logging to dockerng.login

This can be used with debug logging enabled to troubleshoot which
registries this function is attempting to authenticate.

twangboy committed:
Change repo_ng to repo-ng

gtmanfred committed:
require large timediff for ipv6 warning

Commit:
Import datetime

Commit:
Update aws.rst - add Debian default username

terminalmage committed:
Fix regression when CLI pillar override is used with salt-call

terminalmage committed:
Add integration test for 42116

terminalmage committed:
Fix pillar.get when saltenv is passed

Ch3LL committed:
Update builds available for rc1

Commit:
Merge pull request #42142 from Ch3LL/change_builds

Update builds available for rc1

twangboy committed:
Fix unit tests for test_pip

twangboy committed:
Fix problem with handling REG_QWORD in list values

phsteve committed:
Fix docs for puppet.plugin_sync so code-block renders properly and sync is spelled consistently

Commit:
Merge pull request #42155 from phsteve/doc-fix-puppet

Fix docs for puppet.plugin_sync

vutny committed:
Utils: add example of module import

vutny committed:
Fix #42115: parse libcloud "rc" version correctly

Ch3LL committed:
Fix kerberos create_keytab doc

Commit:
Merge pull request #42140 from rallytime/bp-42097

Back-port #42097 to 2016.11

Commit:
Merge pull request #42141 from rallytime/bp-42098

Back-port #42098 to 2016.11

Commit:
Merge pull request #42164 from Ch3LL/fix_kerb_doc

Fix kerberos create_keytab doc

Commit:
Merge pull request #42163 from vutny/fix-42115

Fix #42115: parse libcloud "rc" version correctly

Commit:
Merge pull request #42094 from terminalmage/quiet-exception

Prevent command from showing in exception when output_loglevel=quiet

Commit:
Merge pull request #42121 from terminalmage/issue42114

Fix pillar.get when saltenv is passed

Commit:
Merge pull request #42119 from terminalmage/issue42116

Fix regression in CLI pillar override for salt-call

Commit:
Merge branch '2016.3' into '2016.11'

No conflicts.

Commit:
Merge pull request #42095 from terminalmage/docker-login-debugging

Add debug logging to dockerng.login

Commit:
Fix some documentation issues found in jinja filters doc topic

These doc issues were all found in issue #42151. This commit takes
care of all of the issues presented in #42151 except for the last
task.

Commit:
Merge pull request #42176 from rallytime/bp-42109

Back-port #42109 to 2016.11

Commit:
Merge pull request #42172 from rallytime/merge-2016.11

[2016.11] Merge forward from 2016.3 to 2016.11

garethgreenaway committed:
Back porting the fix for 2017.7 that ensures the order of the names parameter.

twangboy committed:
Disable tests that require pwd in Windows

Commit:
Merge pull request #42175 from rallytime/bp-39366

Back-port #39366 to 2016.11

vutny committed:
Add missing doc for ``utils_dirs`` Minion config option

terminalmage committed:
Only pass a saltenv in orchestration if one was explicitly passed (2016.11)

Also, only pass pillarenv if explicitly passed. _get_opts() in the state
module will apply the correct saltenv (if applicable) when no explicit
saltenv/pillarenv is passed.

Commit:
Merge pull request #42173 from rallytime/bp-37424

Back-port #37424 to 2016.11

twangboy committed:
Add missing config to example

Commit:
Merge pull request #42211 from terminalmage/issue40928

Only pass a saltenv in orchestration if one was explicitly passed (2016.11)

Commit:
Merge pull request #42215 from twangboy/win_iis_docs

Add missing config to example

Commit:
Add note about "to_bytes" jinja filter issues when using yaml_jinja renderer

tdutrion committed:
Remove duplicate instruction in Openstack Rackspace config example

AstraLuma committed:
Abolish references to `dig` in examples.

AstraLuma committed:
Make note of dig partial requirement.

Commit:
Merge pull request #42235 from astronouth7303/patch-1-2016.3

Abolish references to `dig` in examples.

gtmanfred committed:
Only use unassociated ips when unable to allocate

When checking the unassociated ips first, in Parallel mode, each server will
use the same one unallocated ip.

UtahDave committed:
simple doc updates

Commit:
Merge pull request #42252 from UtahDave/2016.11local

simple docstring updates

Commit:
Merge pull request #42253 from gtmanfred/2016.11

Only use unassociated ips when unable to allocate

Commit:
Merge pull request #42181 from garethgreenaway/42137_backport_fix_from_2017_7

fixes to state.py for names parameter

Commit:
Some minor doc fixes for dnsutil module so they'll render correctly

Commit:
Update minion restart section in FAQ doc for windows

Fixes #41116

Commit:
Merge pull request #42262 from rallytime/bp-42224

Back-port #42224 to 2016.11

Commit:
Merge pull request #42261 from rallytime/minor-doc-fix

Some minor doc fixes for dnsutil module so they'll render correctly

Commit:
Add some clarity to "multiple quotes" section of yaml docs

Fixes #41721

terminalmage committed:
pkg.installed: pack name/version into pkgs argument

This allows a version of 'latest' to work when just a name and version
is passed.

Commit:
Add mono-spacing to salt-minion reference for consistency

Commit:
Merge pull request #42123 from vutny/fix-master-utils-import

DOCS: describe importing custom util classes

Commit:
Handle libcloud objects that throw RepresenterErrors with --out=yaml

Fixes #42152

Commit:
Add a note about using different styles of quotes.

lubyou committed:
Force file removal on Windows. Fixes #42295

Commit:
Merge pull request #42289 from CorvinM/bp-41543

Multiple empty_password fixes for state.user

Commit:
Merge pull request #41945 from garethgreenaway/41936_allow_identity_files_with_user

Fixes to modules/git.py

Commit:
Add clarification to salt ssh docs about key auto-generation.

Fixes #42267

mirceaulinic committed:
Default skip_verify to False

This has been partially tackled in https://github.com/saltstack/salt/pull/41528,
but exactly the main entry point was missed.

mirceaulinic committed:
Trailing whitespaces

Commit:
Bugfix: Jobs scheduled to run at a future time stay pending for Salt minions (bsc#1036125)

Commit:
Merge pull request #42314 from rallytime/fix-42267

Add clarification to salt ssh docs about key auto-generation.

Commit:
Merge pull request #42308 from lubyou/42295-fix-file-absent-windows

Force file removal on Windows. Fixes #42295

Commit:
Merge pull request #42282 from rallytime/fix-42152

Handle libcloud objects that throw RepresenterErrors with --out=yaml

Commit:
Merge pull request #42269 from rallytime/fix-41721

Add some clarity to "multiple quotes" section of yaml docs

Commit:
Merge pull request #42275 from terminalmage/issue42194

pkg.installed: pack name/version into pkgs argument

Commit:
Merge pull request #42264 from rallytime/fix-41116

Update minion restart section in FAQ doc for windows

twangboy committed:
Detect server OS with a desktop release name

Ch3LL committed:
[2017.7] Update version numbers in doc config for 2017.7.0 release

terminalmage committed:
is_windows is a function, not a propery/attribute

While not fatal, this could potentially cause problems running this
test on Windows.

Commit:
Merge pull request #42353 from terminalmage/fix-git-test

is_windows is a function, not a propery/attribute

Commit:
Merge branch '2016.11' into '2017.7'

Conflicts:
  - salt/grains/core.py
  - salt/modules/git.py
  - salt/modules/pillar.py
  - salt/states/saltmod.py
  - salt/states/user.py
  - tests/integration/states/handle_iorder.py

twangboy committed:
Add note about patched windows packages

twangboy committed:
Remove build and dist directories before install

twangboy committed:
Add blank line after delete

Commit:
Merge pull request #42179 from rallytime/fix-42151

Fix some documentation issues found in jinja filters doc topic

Commit:
Merge pull request #42327 from mirceaulinic/patch-10

Default skip_verify to False

Commit:
Merge pull request #42340 from isbm/isbm-jobs-scheduled-in-a-future-2017.7-bsc1036125

Bugfix: Jobs scheduled to run at a future time stay pending for Salt …

Commit:
Merge pull request #42363 from rallytime/merge-2017.7

[2017.7] Merge forward from 2016.11 to 2017.7

Commit:
Merge pull request #42361 from Ch3LL/doc-update-2017.7

[2017.7] Update version numbers in doc config for 2017.7.0 release

Commit:
Merge pull request #42364 from twangboy/release_notes_2017.7.0

Windows Package notes for 2017.7.0

Ch3LL committed:
Add initial 2017.7.1 Release Notes File

Commit:
Merge pull request #42182 from twangboy/win_unit_test_useradd

Fix `unit.modules.test_useradd` for Windows

Commit:
Merge pull request #42154 from twangboy/win_unit_test_reg

Fix `unit.modules.test_reg_win` for Windows

Commit:
Merge pull request #42150 from twangboy/win_unit_test_pip

Fix `unit.modules.test_pip` for Windows

Commit:
Merge pull request #42373 from Ch3LL/add_2017.7.1

Add initial 2017.7.1 Release Notes File

Commit:
Merge pull request #42366 from twangboy/win_fix_build

Remove build and dist directories before install

Commit:
Merge pull request #42347 from twangboy/win_fix_ver_grains

Fixes problem with Version and OS Release related grains on certain versions of Python

twangboy committed:
Update Python and other reqs

Commit:
Update old "ref" references to "rev" in git.detached state

Fixes #42381

The "rev" kwarg was added to replace "ref" in #38898, however,
when switching the state over to "rev", some stacktraces occur
due to some remaining "ref" references.

CorvinM committed:
Fix documentation misformat in salt.states.file.replace

twangboy committed:
Add support for Py3

Commit:
Merge pull request #42408 from CorvinM/file-replace-doc-fix

Fix documentation misformat in salt.states.file.replace

seedickcode committed:
Fix file.managed check_cmd file not found - Issue #42404

twangboy committed:
Fix hard coded pip path

twangboy committed:
Remove build and dist, sign pkgs

terminalmage committed:
Only force saltenv/pillarenv to be a string when not None

This fixes a regression introduced to make numeric saltenv/pillarenv
work properly.

terminalmage committed:
Modify our custom YAML loader to treat unicode literals as unicode strings

garethgreenaway committed:
Updating the versions function inside the manage runner to account for when a minion is offline and we are unable to determine it's version.

Commit:
Merge pull request #42433 from terminalmage/issue42403

Only force saltenv/pillarenv to be a string when not None

Commit:
Merge pull request #42409 from twangboy/mac_py3_scripts

Add Scripts to build Py3 on Mac

Commit:
Merge pull request #42411 from seedickcode/check_cmd_fix

Fix file.managed check_cmd file not found - Issue #42404

garethgreenaway committed:
Updating the slack engine in 2017.7 to pass the args and kwrags correctly to LocalClient

salt-jenkins commented:
@ibuil, thanks for your PR! By analyzing the history of the files in this pull request, we identified @twangboy, @toanju and @rallytime to be potential reviewers.

Ch3LL committed:
update windows urls to new py2/py3 naming scheme

terminalmage committed:
Document future renaming of new rand_str jinja filter

The name and the documentation for this filter are entirely inaccurate
for what it actually does. It will be renamed in the Oxygen release.

Commit:
Bump warning version from Oxygen to Fluorine in roster cache

The change was introduced in 2017.7 - we need to provide at least
2 feature releases before removing this warning. Therefore, this
PR bumps the version we will warn until from Oxygen to Fluorine.

Commit:
Merge pull request #42452 from Ch3LL/fix_url_windows

update windows urls to new py2/py3 naming scheme

Commit:
Merge pull request #42454 from terminalmage/jinja-docs-2017.7

Document future renaming of new rand_str jinja filter

garethgreenaway committed:
Merge pull request #42027 from gtmanfred/2017.7

import salt.minion for EventReturn for Windows

garethgreenaway committed:
Merge pull request #42031 from skizunov/develop3

Fix: Reactor emits critical error

garethgreenaway committed:
Merge pull request #42399 from rallytime/fix-42381

Update old "ref" references to "rev" in git.detached state

garethgreenaway committed:
Merge pull request #42435 from terminalmage/issue42427

Modify our custom YAML loader to treat unicode literals as unicode strings

garethgreenaway committed:
Merge pull request #42436 from garethgreenaway/42374_manage_runner_minion_offline

Fixes to versions function in manage runner

Commit:
Merge pull request #42461 from rallytime/bump-roster-cache-deprecations

Bump warning version from Oxygen to Fluorine in roster cache

Commit:
Merge pull request #42444 from garethgreenaway/42357_2017_7_pass_args_kwargs_correctly

[2017.7] Fix to slack engine

rallytime commented:
Hi @ibuil - this still isn't quite right as there are more commits here than just your change. I'm going to close this so tests don't keep starting on this PR as other PRs are merged into Salt's 2017.7 branch. Please let us know if we can help in any way. :)
